codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/transformers/RefactorConstrainedMemberType.kt

```diff
@@ -0,0 +1,195 @@
+package software.amazon.smithy.rust.codegen.server.smithy.transformers
+
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.AbstractShapeBuilder
+import software.amazon.smithy.model.shapes.MemberShape
+import software.amazon.smithy.model.shapes.Shape
+import software.amazon.smithy.model.shapes.ShapeId
+import software.amazon.smithy.model.shapes.StructureShape
+import software.amazon.smithy.model.traits.Trait
+import software.amazon.smithy.model.transform.ModelTransformer
+import software.amazon.smithy.rust.codegen.core.smithy.DirectedWalker
+import software.amazon.smithy.rust.codegen.core.util.orNull
+import software.amazon.smithy.rust.codegen.server.smithy.allConstraintTraits
+import software.amazon.smithy.rust.codegen.server.smithy.hasConstraintTrait
+import software.amazon.smithy.rust.codegen.server.smithy.traits.RefactoredStructureTrait
+import software.amazon.smithy.utils.ToSmithyBuilder
+import java.lang.IllegalStateException
+import java.util.*
+
+object RefactorConstrainedMemberType {
+    private data class MemberTransformation(
+        val newShape: Shape,
+        val memberToChange: MemberShape,
+        val traitsToKeep: List<Trait>,
+    )
+
+    /**
+     * Transforms all member shapes that have constraints on them into equivalent
+     * non-constrained member shapes.
```

Comment:
```suggestion
     * Transforms all structure member shapes that have constraints on them into equivalent
     * non-constrained structure member shapes targeting synthetic constrained structure shapes with the member's constraints.
```

I'd move this documentation above `object` and include a simple example Smithy model so that the reader can get the gist of this transformation.

E.g.:

```
structure A {
    @length(min: 1, max: 69)
    overriddenConstrainedString: ConstrainedString
}

@length(min: 2, max: 10)
@pattern("^[A-Za-z]+$")
string ConstrainedString
```

to

```
structure A {
    overridenConstrainedString: OverriddenConstrainedString
}

@length(min: 1, max: 69)
@pattern("^[A-Za-z]+$")
OverriddenConstrainedString
```